lib/game.rb

```diff
@@ -0,0 +1,90 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+  ]
+
+  @@all = {"X"=>0, "O"=>0, "Senseless Destruction"=>0}
+
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+    puts "\nSweet. When it's your turn, enter a number 1 through 9 to place your token \non the corrosponding spot of the board, '1' being the top left corner and '9' being the bottom right.\n\n"
+  end
+
+  def current_player
+    if self.board.turn_count.even?
+      self.player_1
+    else
+      self.player_2
+    end
+  end
+
+  def over?
+    draw? || won?
+  end
+
+  def draw?
+    self.board.full? && !won?
+  end
+
+  def won?
+    cells = self.board.cells
+    WIN_COMBINATIONS.find do |combo|
+      self.board.taken?(combo[0]+1) && cells[combo[0]] == cells[combo[1]] && cells[combo[0]] == cells[combo[2]]
+    end
+  end
+
+  def winner
+    won? ? self.board.cells[won?[0]] : nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.